aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -682,9 +682,9 @@ std::tuple<Tensor, Tensor> _scaled_dot_product_attention_math(
     attn = at::softmax(attn, -1);
     if (dropout_p > 0.0) {
       if (dropout_mask.has_value()) {
-        auto attn_dropout_masked = attn.masked_fill(dropout_mask->logical_not(), 0.0);
+        attn = attn.masked_fill(dropout_mask->logical_not(), 0.0);
```

Comment:
Wait doesn't this change the behavior?

```
auto attn_dropout_masked = attn.masked_fill(dropout_mask->logical_not(), 0.0);
[...]
return std::make_tuple(at::matmul(attn_dropout_masked, value * dropout_scaling), attn);
```

vs.

```
attn = attn.masked_fill(dropout_mask->logical_not(), 0.0);
[...]
return std::make_tuple(at::matmul(attn, value * dropout_scaling), *attn*);
```

You're using a different right hand operand in the second version. Is this fixing a bug?

Reply:
Ah, never mind. dropout_mask is just a debug feature and you're updating it for testing.